lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py

```diff
@@ -361,19 +361,6 @@ def test_coefficients_from_mean_non_standard_units(self):
         self.assertEqual(
             self.plugin.coeff_names, expected_coeff_names)
 
-    @ManageWarnings(
```

Comment:
The situation of a cube with all of time, forecast reference time and forecast period associated with the same dimension should never happen.  I couldn't figure out how it was possible to concatenate cubes with this dimensionality - every test case I tried failed.  I suspect the only reason this ever worked was because the original developer set up test cubes with incorrect metadata (ie where the time, forecast period and forecast reference time were self-inconsistent).  If you set up a cube with realistic values of these coordinates it cannot associate them all with the same dimension, because mathematically they must describe exactly 2 dimensions.

An alternative way to set up a test for multiple forecast periods would be to set up a "historic" cube of forecasts from the same cycle.  If there's a need for this test I can do that.  However the ensemble calibration test setup functions are inflexible and horrible to work with, so I didn't want to put a lot of effort into that if it wasn't necessary.

Reply:
The point of this test was to check the warning was raised, which is now not tested. When I asked for the test to be added I was thinking of the case where someone had passed in historic forecasts at the same validity times, but from cycles that meant they had different forecast periods. I think you're telling me that this kind of situation will fail when you try to merge the historic forecasts together. If so then it is ok to remove the test but we should also remove the code that raises the warning. 

It sounds like we need to upgrade these unit tests to use the new setup cubes functionality fairly urgently, especially as we are changing this code a lot at the moment.

Reply:
As discussed, I'll remove the warning raised in the code.  If this situation occurs, the ensemble calibration CLI will fail to load the forecasts, `load_cube` calls `merge_cube` and this should raise a MergeError.  If it doesn't, the inputs should be checked there to make sure they have the right number of dimensions.

Reply:
I've now added a note on this to the relevant JIRA ticket.